doc: enhance breaking change rules (#2343)
AmadeusITGroup/otter: SECURITY.md

Two major versions are released per year:

- A technical version is released during summer which can include:
  - Upgrades of the Otter library peer dependencies, such as:
    - Angular
    - Typescript
    - RxJS
    - NgRx
    - etc...

>   - Breaking changes covered by `ng update` code patching

Reviewer: We should ensure that `ng update` will fully cover the breaking changes 
sometimes we are doing `ng update` script that match only the default behavior

Two major versions are released per year:

- A technical version is released during summer which can include:
  - Upgrades of the Otter library peer dependencies, such as:
    - Angular
    - Typescript
    - RxJS
    - NgRx
    - etc...
  - Breaking changes covered by `ng update` code patching

>   - breaking change with **very low migration cost** can be exceptionally accepted, the code review will determine its acceptance.

Reviewer: ```suggestion
  - Breaking change with **very low migration cost** can be exceptionally accepted, the code review will determine its acceptance.
```

- Another one to deliver new features of the Otter library, released at the beginning of the year.

## Release version support

The team will provide support on the latest minor version of major versions **N (current)**, **N-1** and **N-2**.

A major version will be supported as long as it relies on a [non deprecated version of Angular](https://angular.io/guide/releases#support-policy-and-schedule).

> > **Note**: It is also important to note that Otter supports the same browser versions as Angular.

Reviewer: ```suggestion
> [!NOTE]
> It is also important to note that Otter supports the same browser versions as Angular.
```

Two major versions are released per year:

- A technical version is released during summer which can include:
  - Upgrades of the Otter library peer dependencies, such as:
    - Angular
    - Typescript
    - RxJS
    - NgRx
    - etc...
  - Breaking changes covered by `ng update` code patching

>   - Breaking change with **very low migration cost** can be exceptionally accepted, the code review will determine its acceptance.

Reviewer 2: ```suggestion
  - Breaking changes with a **very low migration cost** can be exceptionally accepted, the code review will determine its acceptance.
```
